sdk-incubator-lib/src/androidTest/kotlin/cash/z/ecc/android/sdk/internal/DerivationToolImplTest.kt

```diff
@@ -26,19 +27,45 @@ class DerivationToolImplTest {
                     network = network,
                     account = account,
                 )
-            assertEquals(Base64.encode(key), "byyNHiMfj8N2tiCHc4Mv/0ts0IuUqDPe99MvW8B03IY=")
+            assertEquals("byyNHiMfj8N2tiCHc4Mv/0ts0IuUqDPe99MvW8B03IY=", Base64.encode(key))
         }
 
     @OptIn(ExperimentalEncodingApi::class)
     @Test
-    fun testDerivedArbitraryWalletKey() =
+    fun testAliceDerivedArbitraryWalletKeyInBase64() =
         runTest {
             val key =
                 DerivationTool.getInstance().deriveArbitraryWalletKey(
                     contextString = CONTEXT.toByteArray(),
                     seed = seedPhrase.toByteArray(),
                 )
-            assertEquals(Base64.encode(key), "1xm/qCXZqXgiRFT1IVk97+5gv9BE5AjkOVWHwYU9RYQ=")
+            assertEquals("1xm/qCXZqXgiRFT1IVk97+5gv9BE5AjkOVWHwYU9RYQ=", Base64.encode(key),)
+        }
+
+    @OptIn(ExperimentalStdlibApi::class)
+    @Test
+    fun testVectorDerivedArbitraryWalletKey() =
+        runTest {
+            val secretKey =
+                DerivationTool.getInstance().deriveArbitraryWalletKey(
+                    contextString = "5a63617368207465737420766563746f7273".hexToByteArray(),
```

Comment:
For at least one of the two test vectors, can you use `"Zcash test vectors"` converted into a byte array? That's how Zashi Android is constructing its context string, and I want to ensure we test that this conversion works the same way in Kotlin and Rust.

(or both, and test with both this and hex)

Reply:
This is what I mean:
```suggestion
                    contextString = "Zcash test vectors".toByteArray(),
```

Reply:
I want the above change specifically to ensure that this code in Zashi Android is not breaking its specification by silently encoding the string in an unexpected byte format: https://github.com/Electric-Coin-Company/zashi-android/pull/1643/files#diff-055de5dc228c0e87d964a8ec4865d3e48d0c20067cc3e6e03061d096cf481ec5R69